cmd/kubeadm/app/cmd/init.go

```diff
@@ -163,6 +163,24 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 		if len(initRunner.Options.SkipPhases) == 0 {
 			initRunner.Options.SkipPhases = data.cfg.SkipPhases
 		}
+		// dump DNS and Proxy status by skip-phases
+		if isStringInSlice("addon/coredns", initRunner.Options.SkipPhases) {
+			data.cfg.DNS.Disabled = true
+		}
+		if isStringInSlice("addon/kube-proxy", initRunner.Options.SkipPhases) {
+			data.cfg.Proxy.Disabled = true
+		}
+		if isStringInSlice("addon", initRunner.Options.SkipPhases) {
+			data.cfg.DNS.Disabled = true
+			data.cfg.Proxy.Disabled = true
+		}
+		// If DNS or Proxy is disabled, skip the corresponding phase.
+		if data.cfg.DNS.Disabled {
+			initRunner.Options.SkipPhases = append(initRunner.Options.SkipPhases, "addon/coredns")
+		}
+		if data.cfg.Proxy.Disabled {
+			initRunner.Options.SkipPhases = append(initRunner.Options.SkipPhases, "addon/kube-proxy")
+		}
```

Comment:
this will cause a double append - having the `addon/* `phases in `SkipPhases` two times?
this logic should be after:
```
if len(initRunner.Options.SkipPhases) == 0 {
	initRunner.Options.SkipPhases = data.cfg.SkipPhases
}
```
before the logic to update DIsabled fields?

Reply:
+1, this suggestion by @neolit123 is that if it's set by the CLI flags, then just use it otherwise read if from cfg.